### DISCUSSION TITLE
Minimum generation level

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#107] [Q&A: Data]
Hi! I got lost while browsing the input data in "minloadfrac0.csv" (below). Does this input file represent the minimum generation level adjustment to calculate the curtailment of the generator? If so, is there a minimum generation level adjustment for gas-ct and gas-cc as well? Thank you in advance!

![image](https://user-images.githubusercontent.com/78582696/157597113-431a1a7a-12c3-43b0-9db3-a6dd518722f6.png)

**Comment by vcelloho** [comment]
This represents the minimum level of operations required in each time slice within a season. The intention is to represent the inflexibility of a generator that would have challenges with startups during a short period of time. ReEDS time slice generate a representative day made up of four or five time slices per season. While using a full hourly representation a coal plant could shut down and be restarted this isn't feasible in the timescale of ReEDS.  For this reason we include a combination of constraints that require the covered inflexible generators to operate above a minimum level or be off for the entire season.

Natural Gas plants having faster start times are allowed to turn off for a time slice and as a result are not included in this dataset.

https://github.com/NREL/ReEDS_OpenAccess/blob/023c9c197b1c5af77e659fa0d1d1625701aa587f/c_supplymodel.gms#L913

**Reply by wanyingw193** [reply]
Thank you for your answer! Could I ask where the data source for the minimum generation level is? Is it the same source as the table 25 of [ReEDS 2021 tech doc](https://www.nrel.gov/docs/fy21osti/78195.pdf)? 

I was also wondering what size of startup time can be classified as faster startup time? I may have missed the file that sets the start time for each technology, is it in ramprate.csv? 

Additionally, is the minimum generation capacity related to minCF.csv? In my understanding, the minimum capacity factor will indirectly affect the minimum generation level?

Thank you so much!

**Comment by wanyingw193** [comment]
Hi, thank you for your attention!

(1) What are the different applications between the input files minCF and minloadfrac0? 
(2) Do minCF (GSw_minCF) and minloadfrac0 (Sw_mingen) always need to be turn-on together? 
(3) Do Sw_mingen in case_test.csv and B1_osprey.py need to be turn-on together to include minimum generation requirements?

**Comment by wesleyjcole** [comment]
1. minCF set a minimum annual capacity factor for a given technology while minloadfrac0 sets a minimum turndown level for a plant.
2. They need not be turned on together, as they play different roles.  The minCF ensures a minimum level of operation while the minloadfrac0 reduces the flexibility of plants.
3. These need not be turned on together.  We typically do not turn on the mingen switch in B1_osprey as it slows down run time without meaningfully impacting the results.

**Comment by wanyingw193** [comment]
Hi, can I understand in this way? If the mingen switch of B1_osprey is turn off, then the partial curtailment due to the level of Mingen will no longer take place. Therefore, the minimum generation capacity is only determined by miniloadfrac and GEN, and it will not affect the Objective function. 

Are there any other parameters that would also affect the MINGEN variable? Thanks!

**Reply by wesleyjcole** [reply]
If the mingen switch in B1_osprey is turned off, then the hourly dispatch will not apply the mingen constraints.  We do apply a post-processing mingen adjustment after B1_osprey runs to increase curtailment in hours where generation is below the allowable mingen levels.

Within the ReEDS model (c_supplymodel) the MINGEN variable impacts the curtailment calculations (increasing MINGENraises curtailment while reducing MINGEN increases curtailment).  The equation eq_minloading is what requires GEN to not go below minloadfrac over the course of a day.
